### ISSUE TITLE
custom_components.hacs.exceptions.HacsException: No manifest.json file found '/manifest.json'

### ISSUE THREAD

**Issue by grinco** [grinco/ote_rate#16]
I can't wrap my head around why hacs keeps complaining that manifest.json is missing. @yardee did you have any similar issues when you changed the directory structure?

Also there are a few things on the todo: fix github actions, especially the build and upload zip file process. I have no experience with it so it might take a little while before I figure it out.

**Comment by grinco** [comment]
https://github.com/grinco/ote_rate/releases/tag/v0.2.2 is marked as beta for now until update is fixed.

**Comment by grinco** [comment]
Looks like it has something to do with hacs caching. If I set v0.2.2 as latest, remove repository and re-add it hacs - it's updating just fine. I'll try to downgrade and update again. If it works - will resolve this one.

**Comment by grinco** [comment]
Yup, I was able to validate the suspected behavior by downgrading to  https://github.com/grinco/ote_rate/releases/tag/v0.2.0 and back to v.0.2.2. 

TL;DR: looks like HACS doesn't handle well changing "[content_in_root](https://github.com/grinco/ote_rate/pull/15/files#diff-2004f3033aaa650b74a155b56a1e6110e85583824c83f3cb5a0c2856bd00483f)" between releases. To work around it one must remove the repository from HACS and re-add, or install the update manually. I'll collect all necessary logs and report a bug against HACS.

**Comment by yardee** [comment]
I have no issues with upgrading to new structure, but it might be just thanks to I had to change the respository to my fork anyway. Build action looks good now right?

**Comment by grinco** [comment]
For some reason it's not triggered. See, I have just published https://github.com/grinco/ote_rate/releases/tag/v0.2.3 and the build workflow didn't trigger. I'll give it a try on a dummy repo and will fix when I have time. For now I've updated the ZIP manually and attached to the release.

**Comment by grinco** [comment]
Found the issue with the build process. It only works when the release is published using the CLI.

**Comment by yardee** [comment]
Thats strange, I have tried it on my fork and it triggered successfully: https://github.com/yardee/ote_rate/actions/workflows/build.yaml (just needed to set rights to write)

**Comment by yardee** [comment]
Also I found out checks on pull requests are not triggered. Like here 
https://github.com/yardee/ote_rate/pull/9/checks
https://github.com/yardee/ote_rate/actions/workflows/pull.yml

**Comment by yardee** [comment]
I think I have just setup Branch protection rules in the repository settings

**Comment by grinco** [comment]
It's because your pull request introduced the actions - it didn't exist in the repo before. It works now. Here's an example:
https://github.com/grinco/ote_rate/actions/runs/7604215259
https://github.com/grinco/ote_rate/pull/18/checks

**Comment by yardee** [comment]
Sure, I ment on this [PR](https://github.com/grinco/ote_rate/pull/17), where were pull request actions already merged. But it doesn't matter, it is working now. Thanks
